### PR TITLE
[tree view][docs] Adjust all demos height to account for the updated design

### PR DESCRIPTION
**Pull request by danilo-leal**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Closes https://github.com/mui/mui-x/issues/11976

Follow-up on the design update done on https://github.com/mui/mui-x/pull/11529 😃 I haven't added any `overflow` and rather just added fixed height for demos with a short amount of items and a min-height for those that have a bigger list.

### CODE REVIEW

**Comment by danilo-leal** [comment]
Uhm... the test_types check is picking up components I didn't even touch on this PR. 🤔 Any ideas why?

**Comment by flaviendelangle** [comment]
No idea, you can try to merge with `next`

File: docs/data/data-grid/editing/ValidateServerNameGrid.js [hunk 1]
```diff
@@ -23,13 +23,10 @@ function validateName(username) {
   const existingUsers = rows.map((row) => row.name.toLowerCase());
 
   return new Promise((resolve) => {
```
Comment (flaviendelangle) [inline docs/data/data-grid/editing/ValidateServerNameGrid.js:25]: No idea why this formatting change, if you run `yarn prettier` does it revert it?

Comment (alexfauquette) [reply docs/data/data-grid/editing/ValidateServerNameGrid.js]: running `yarn prettier` modifies some files. let see if it fixes the CI

Comment (danilo-leal) [reply docs/data/data-grid/editing/ValidateServerNameGrid.js]: Yeah, similarly with `pnpm docs:typescript:formatted` — it was picking up files I hadn't touched on this PR, so I just avoided running those.

Comment (noraleonte) [reply docs/data/data-grid/editing/ValidateServerNameGrid.js]: Could have been some prettier settings 🤔 Should be fine now I think

Comment (danilo-leal) [reply docs/data/data-grid/editing/ValidateServerNameGrid.js]: Sweet! Thanks, y'all! Would appreciate a review now :)

**Review by noraleonte — APPROVED** [review]
LGTM 🎉

**Pull request merged**